src/gluonts/dataset/parallelized_loader.py

```diff
@@ -84,16 +85,77 @@ def reduce_ndarray(data):
 ForkingPickler.register(nd.NDArray, reduce_ndarray)
 
 
+def _is_stackable(
+    arrays: List[Union[np.ndarray, mx.nd.NDArray, Any]],
+) -> bool:
+    """
+    Check if elements are scalars, have too few dimensions, or their
+    time axes have equal length; i.e. they are directly `stack` able.
+    """
+    if isinstance(arrays[0], (mx.nd.NDArray, np.ndarray)):
+        s = set(arr.shape[0] for arr in arrays)
+        return len(s) <= 1 and arrays[0].shape[0] != 0
+    return True
+
+
+def _pad_arrays(
+    data: List[Union[np.ndarray, mx.nd.NDArray]], pad_axis: int = 0,
+) -> List[Union[np.ndarray, mx.nd.NDArray]]:
+    assert isinstance(data[0], (np.ndarray, mx.nd.NDArray))
+    is_mx = isinstance(data[0], mx.nd.NDArray)
+
+    # MxNet causes a segfault when persisting 0-length arrays. As such,
```

Comment:
So I'm wondering, why does it come to this, i.e. why would we want to persist 0-length arrays? Can we not just pad the whole length with 0s or whatever the padding value is? 
What about removing that entry from the batch?

Reply:
Persisting is not really relevant here. Under normal circumstances we do pad 0-length arrays to the appropriate length. This line only kicks in when the *entire batch* has 0-length, in which case we pad to a dummy dimensionality of 1.

Unfortunately the entry is valid data. You want to know if you observed nothing--can't kick it out of the batch.

Reply:
I see.